connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2896,6 +2896,16 @@ object functions {
    */
   def levenshtein(l: Column, r: Column): Column = Column.fn("levenshtein", l, r)
 
+  /**
+   * Computes the Levenshtein distance of the two given string columns if it's less than or
+   * equal to a given threshold.
+   * @return result distance, or -1
```

Comment:
When any argument of the xxx function is null, it will return null.
BTW, function levenshtein inherits NullIntolerant.
https://github.com/apache/spark/blob/5ec138546205ba4248cc9ec72c3b7baf60f2fede/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L2150-L2156

Reply:
So when does it return -1?

Reply:
When the `actual` Levenshtein distance between the two given strings > `threshold`, it return -1.

Reply:
```
scala> df.select($"l", $"r", levenshtein($"l", $"r"), levenshtein($"l", $"r", 2), levenshtein($"l", $"r", 3)).show()
+------+-------+-----------------+--------------------+--------------------+
|     l|      r|levenshtein(l, r)|levenshtein(l, r, 2)|levenshtein(l, r, 3)|
+------+-------+-----------------+--------------------+--------------------+
|kitten|sitting|                3|                  -1|                   3|
|  frog|   NULL|             NULL|                NULL|                NULL|
+------+-------+-----------------+--------------------+--------------------+

```

it seems it returns -1 if inputs are not-null and the disttance > threshold.

but we don't have a test for nullability in `StringFunctionsSuite`

Reply:
Let me add UT for it?